// @middy/validator/transpile.cjs
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    transpileSchema: function() {
        return transpileSchema;
    },
    transpileLocale: function() {
        return transpileLocale;
    }
});
const _compile = _interop_require_default(require("ajv-cmd/compile"));
const _ftl = _interop_require_default(require("ajv-cmd/ftl"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
const ajvDefaults = {
    strict: true,
    coerceTypes: 'array',
    allErrors: true,
    useDefaults: 'empty',
    messages: true
};
const transpileSchema = (schema, ajvOptions)=>{
    const options = {
        ...ajvDefaults,
        ...ajvOptions
    };
    return (0, _compile.default)(schema, options);
};
const transpileLocale = _ftl.default;
